Post-workshop additions: Support forms

EthanThatOneKid opened the pull request:
During the 3rd Hot OSS Summer workshop, the API workshop, I mentioned that it is possible to use native HTML forms to send and retrieve data from the client and server.

- Implements a native, client-side rendered list of HTML forms for each TODO in memory.
- Introduces the `DB` interface and two implementations: `InMemoryDB` and `DenoKvDB`.
- Introduces `.github/workflows/check.yaml` GitHub workflow.

### Future work

- Create a version of `main.ts` that uses `DenoKvDB`; ready for deployment on [Deno Deploy](https://deno.com/deploy/docs).

EthanThatOneKid committed:
WIP

EthanThatOneKid committed:
Add `deno tasks start` task

EthanThatOneKid committed:
Update todo.ts

EthanThatOneKid committed:
Support HTML form requests (WIP)

oscisn93 commented on main.ts:
```diff
@@ -13,64 +16,85 @@ function main() {
   switch (true) {
     case request.method === "GET" && url.pathname === "/": {
       const body = Deno.readTextFileSync("static/index.html");
+      console.log("Found index.html");
```
shouldn't all of these be removed?

EthanThatOneKid replied:
> shouldn't all of these be removed?

Good question! I was originally considering leaving them for the ability to see how the server works as a learning tool. What do you think, @oscisn93? Another option is to properly demo [Deno’s standard logger](https://deno.land/std/log/mod.ts).

oscisn93 replied:
Yeah I think the standard logger is a great idea. And a good teaching moment

EthanThatOneKid replied:
Thanks for your input! Here's what I did: <https://github.com/acmcsufoss/api-workshop/pull/2/commits/db1e53f808143aff4b1bdc95c30906cc31fc7d48>.

oscisn93 commented on todo.ts:
```diff
@@ -10,8 +10,7 @@ export interface Todo {
-  const { 2: id } = url.split("/");
-  console.log({ url, id });
+export function parseTodoID(pathname: string): string {
+  const { 2: id } = pathname.split("/");
```
maybe a bit clever with the de-structuring here?
I feel like i had to work a bit harder than necessary to figure out you're asking for the second element in the array/split string.

EthanThatOneKid replied:
I hope this is more clear: <https://github.com/acmcsufoss/api-workshop/pull/2/commits/e922f5c1ee58d9eaa00529f7830316ea808dda30>.

EthanThatOneKid replied:
Alternatively, I could keep the `const { 2: id } = pathname.split("/");` but with a new comment:
> Extracts the value at index 2 from the array returned by splitting the 'pathname' string using "/" as the delimiter and assigns it to the variable 'id'.
>
> See:
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

oscisn93 reviewed (changes requested):
Just a few small inputs. Overall thought great work and great workshop!

EthanThatOneKid committed:
Replace `console.log`s with standard logger

In response to this review comment: <https://github.com/acmcsufoss/api-workshop/pull/2#discussion_r1248360722>.

EthanThatOneKid committed:
Add `acmcsuf.com` favicon

EthanThatOneKid committed:
Improve readability of `parseTodoID` helper

In response to this review comment: <https://github.com/acmcsufoss/api-workshop/pull/2#discussion_r1248362116>.
